app/models/katello/concerns/organization_extensions.rb

```diff
@@ -197,7 +197,7 @@ def parent_id=(_parent_id)
 
         def latest_repo_discovery
           ForemanTasks::Task::DynflowTask.for_action(::Actions::Katello::Repository::Discover)
-            .for_resource(::User.current).order("started_at").last
+            .where(user: ::User.current).order("started_at").last
```

Comment:
@adamruzicka I see that for_resource is looking for Links against the given task type, but Repository::Discover tasks has no Links so nothing was returned here. Was that an intentional change with the locking rework in foreman tasks? Do we need to set an action_subject, or is using `where user` appropriate here?

Reply:
Thank you Adam!